docs/topics/patterns/instant-messaging.mdx

```diff
@@ -0,0 +1,359 @@
+        break;
+    }
+  });
+  ```
+  </TabItem>
+</Tabs>
+
+## Security
+
+Momento provides a [robust authorization mechanism](../develop/authentication/tokens) that limits consumers to specific resources and actions. When building an instant messaging feature, it is incredibly important to take scope into consideration so users do not intercept messages they shouldn't have access to.
+
+The pattern described above uses both cache and topic resources. Following the best practice recommended above by making our chat room the topic name, we can create policies in a couple of different ways.
+
+### Client-side only
+
+It's possible to build instant messaging completely on the client side, meaning directly in the browser or mobile application. The policy below would be used in your clients.
+
+```json
+{
+  "permissions": [
+    {
+      "role": "readwrite",
+      "cache": "instant-messaging",
+      "item": {
+        "key": "{chatid}"
+      }
+    },
+    {
+      "role": "publishsubscribe",
+      "cache": "instant-messaging",
+      "topic": "{chatid}"
+    }
+  ]
+}
+```
+
+This policy grants read and write permissions to a specific cache item. The item could be either a sorted set or a list, based on your implementation. The policy does not change based on the data type.
+
+It also grants publish and subscribe permissions to a single topic. Consumers will be able to publish messages to and recieve messages from the one topic. Attempting to subscribe to any other topic will result in an authorization error.
+
+Note that both the cache item and topic share the same value. This is the recommended best practice and the data from the cache item will not interfere with the messages being published to the topic.
+
+:::info
+The tradeoff for a client-side only approach is trust in your front-end. Since this opens write permissions to the client, you must be careful with what you expose in the code. Allowing end users to do write operations requires trust in their usage, since you cannot moderate messages before accepting them. This also takes a distributed approach to storing data instead of centralizing storage in server-side code.
+
+If that is tolerable for your application, it results in unrivaled time-to-market and minimal complexity in your application.
+:::
+
+### Server-side and client-side
+
+If you centralize writes in your server-side code, the pattern changes slightly. Raw message input would be submitted through one topic and be picked up by the server code. Any transformations or moderation would be performed before sending it back down to the verified topic. Below is an example of the policy that would be used on the client side following this pattern.
+
+```json
+{
+  "permissions": [
+    {
+      "role": "read",
+      "cache": "instant-messaging",
+      "item": {
+        "key": "{chatid}"
+      }
+    },
+    {
+      "role": "publishonly",
+      "cache": "instant-messaging",
+      "topic": "{chatid}-raw"
```

Comment:
raw is a reasonable word here, but wondering if it might be a little more clear if you said `unmoderated` or `input` or something.